File: rust_sample/text_highlighter/src/character.rs
use crate::link::LinkType;

#[derive(Debug, Eq)]
pub struct Character {
    pub index: u32,
    pub value: str,
    pub link: LinkType
}

impl Character {
    pub fn is_empty(&self) -> bool {
        self.value == ""
    }
}

impl Character {
    pub fn to_str(&self) -> &str {
        if self.value == "EOF" {
            return ""
        }
        &self.value
    }
}

impl PartialEq for Character {
    fn eq(&self, other: &Self) -> bool {
        self.value == other.value
    }
}

File: nlp_sample/inverted_index_highlighter/schema/character.py
from dataclasses import dataclass

from .link import LinkType

@dataclass
class Character:
    index: int
    value: str  # character
    link: LinkType = LinkType.NONE

    def is_empty(self) -> bool:
        return self.value == ""

    def __eq__(self, other: "Character") -> bool:
        return self.value == other.value

    def __str__(self) -> str:
        if self.value == "EOF":
            return ""
        return self.value
